app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala

```diff
@@ -1629,14 +1629,21 @@ case class Config(
     command: CliCommand = CliCommand.NoCommand,
     network: Option[NetworkParameters] = None,
     debug: Boolean = false,
-    rpcPort: Int = 9999
-)
+    rpcPortOpt: Option[Int] = None) {
+
+  val rpcPort: Int = rpcPortOpt match {
+    case Some(port) => port
+    case None       => command.defaultPort
+  }
+}
 
 object Config {
   val empty: Config = Config()
 }
 
-sealed abstract class CliCommand
+sealed abstract class CliCommand {
+  def defaultPort: Int = 9999
```

Comment:
Nice! I like this design. 

Let's just leave the defaultPort undefined in the superclass and make all the subprojects define it. 

Basically any server project we have is going to require to have a unique rpc port

Reply:
My understanding is this won't be able to be read from a configuration file as the cli doesn't interact with `bitcoin-s.conf`, is that right?

Reply:
> My understanding is this won't be able to be read from a configuration file as the cli doesn't interact with bitcoin-s.conf, is that right?

Correct